lib/releaselib.sh

```diff
@@ -680,17 +680,13 @@ release::gcs::locally_stage_release_artifacts() {
 
   logecho "- Hashing files in ${gcs_stage##$build_output/}..."
   find "$gcs_stage" -type f | while read -r path; do
-    for bits in "256" "512"; do
-      sum="$(common::sha "$path" "$bits" "full")" || return 1
-      echo "$sum" > "${path}.sha${bits}"
-    done
-  done
+    common::md5 $path > "$path.md5" || return 1
+    common::sha $path 1 > "$path.sha1" || return 1
```

Comment:
I suggest we condition these two lines to run only if RELEASE_BRANCH is release-1.13, release-1.14, or release 1.15 and open an issue to remove this small block in six months when 1.15 goes out of support.

Reply:
That doesn't seem like a reasonable timeline to deprecate. We should announce now and then remove in some future release.

https://github.com/kubernetes/release/pull/849#issuecomment-524478348

Reply:
this also costs very little to do, versus it may be painful to unbreak things that consume it.

Reply:
> I don't think checksums are documentation. These are quite literally in the release artifacts and are a thing machine tooling depends on.

I don't disagree in the abstract.  Quality checksums should be dependable.  In the case of our release process implementation, and in the case md5/sha1, I think machine tooling depending on many of our artifacts to be more than informational can become dangerous.

Reply:
Also I really question where the actionable information here will flow.  Will the machine tooling read a deprecation notice?  Will machine tooling owners read a deprecation notice?  Are we actually minimizing any real world breakage, or just pushing it and associated fixes into the arbitrary future on the hope less breakage will happen.  I think in this specific case we're giving the impression of applying an engineering change management process to replace hope and breakage but in a way that ultimately still just amounts to hope and breakage.

Reply:
md5 is a pretty reasonable_non-security_ integrity check still, it's perfectly fine for checking if bits were flipped.

Reply:
we know at least one of them has and sent patches to move off of it (kops / @justinsb)

in any case, if we wish to remove these, we should create an issue to track deprecation, add it to the release notes as deprecated and announce when it will be removed. not just suddenly remove it.